docs/src/compatibility.md

```diff
@@ -104,3 +104,12 @@ PkgB = "≥ 1.2.3"  # [1.2.3,  ∞)
 PkgC = "= 1.2.3"  # [1.2.3, 1.2.3]
 PkgD = "< 1.2.3"  # [0.0.0, 1.2.2]
 ```
+
+### Hyphen specifiers
+
+Hyphen syntax can also be used to specify version ranges:
+
+```toml
+[compat]
+PkgA = "0.1 - 0.2.3"   # [0.1.0,  0.2.3]
```

Comment:
This needs more docs about what happens when the upper bound does not specify a patch or minor version, e.g. `0.1 - 0.2`. Is that equivalent to `[0.1.0 - 0.2.0]`?

Reply:
Good catch. Yes, I should add more docs. And frankly, should also add tests to cover those specific cases.

From Stefan's comment [here](https://github.com/JuliaLang/Pkg.jl/pull/1232#discussion_r294876288), we have:
 
>     * `0.1-0.2.3` means `[0.1, 0.2.3]`
> 
>     * `0.1-0.2` means `[0.1, 0.3)`
> 
>     * `0.1-0` means `[0.1, 1.0)`

I'll add some docs and tests.